Parameterise reboot command

razaj92 opened the pull request:
As per #31, it would be useful to shutdown an instance rather than just reboot, for use cases such as cycling instances in an AWS Autoscaling Groups. 

I have tried to keep it simple by just parameterising the `reboot` value as so you can run any of the `systemctl` commands instead. If you have any other suggestions let me know.

Thanks

mvisonneau commented:
Thanks @razaj92 :)!

yogeek commented:
Hello @awh @mvisonneau ! This feature would be really useful (I have the exact same scenario as @razaj92), is it planned to continue the work on this PR please ?
How can I help ?

evrardjp commented on cmd/kured/main.go:
```diff
@@ -261,9 +264,9 @@ func commandReboot(nodeID string) {
 
 	// Relies on hostPID:true and privileged:true to enter host mount space
-	rebootCmd := newCommand("/usr/bin/nsenter", "-m/proc/1/ns/mnt", "/bin/systemctl", "reboot")
+	rebootCmd := newCommand("/usr/bin/nsenter", "-m/proc/1/ns/mnt", "/bin/systemctl", command)
```
in this case, shouldn't also systemctl be part of the "command" variable? 
In that case the default command could be "/bin/systemctl reboot". Allows more flexibility for non systemd linux systems.

razaj92 replied:
Good point!

renaudhager commented:
Can someone look at this PR?

bboreham commented:
(a) a small conflict with master needs resolved before it will merge
(b) there is a comment from @evrardjp that the design of this feature is wrong

hairyhenderson commented:
@razaj92 are you able to look at this? This'd be a super useful feature 😉

razaj92 commented:
Hey guys 👋 

I've rebased and addressed the comment as requested 🙂 

I am no longer using this anymore myself though, so if anyone would like to test or add to this PR please feel free!

evrardjp commented:
I have added the topic of reboots into our next community meeting, as there are multiple approaches currently opened.
See the agenda here https://docs.google.com/document/d/1bsHTjHhqaaZ7yJnXF6W8c89UB_yn-OoSZEmDnIP34n8/edit# .

Sorry for the delay!

evrardjp commented:
Let's keep that one opened. If we rebase it, we can definitely improve our reboot story here.

I would prefer to see, in this PR, an additional change to expose that new feature in the helm chart.

evrardjp commented:
You might be interested by this PR: https://github.com/weaveworks/kured/pull/297

dholbach commented:
Sorry. Can you please retarget your branch to be merged into `main` please?
